exercises/isbn-verifier/.meta/src/reference/java/IsbnVerifier.java

```diff
@@ -0,0 +1,39 @@
+class IsbnVerifier {
+
+    private String isbn;
+
+    boolean isValid(String s) {
+
+        this.isbn = s.replace("-", "");
+        int total = 0;
+
+        if (isbn.length() != 10) {
+            return false;
+        } else {
+            for (int i = 0; i < isbn.length() - 1; i++) {
+                if(Character.isDigit(isbn.charAt(i))) {
+                    int currentCharVal = Character.getNumericValue(isbn.charAt(i));
+                    total += currentCharVal * (10 - i);
+                } else {
+                    return false;
+                }
+            }
+
+            if (Character.isDigit(isbn.charAt(9))) {
```

Comment:
I would absorb this into the previous loop (just check if `i == isbn.length - 1 && isbn.charAt(i) == 'X'`). But it doesn't matter too much, it's pretty clear either way :) So if you want to keep it separate then that's fine :)

I would use `isbn.length - 1` instead of 9 though, to keep it consistent with the other loop and to prevent the use of "magic" numbers that can confuse anyone reading the code :)